e2e/test_operator/operator_test.go

```diff
@@ -1428,6 +1428,76 @@ var _ = Describe("Operator", Label("e2e", "pr"), func() {
+
+		BeforeEach(func() {
+			if fdbCluster.GetClusterSpec().ReplaceInstancesWhenResourcesChange != nil {
+				Expect(*fdbCluster.GetClusterSpec().ReplaceInstancesWhenResourcesChange).To(BeFalse())
+			}
+			originalPodSpec = fdbCluster.GetPodTemplateSpec(fdbv1beta2.ProcessClassGeneral)
+			originalStoragePodSpec = fdbCluster.GetPodTemplateSpec(fdbv1beta2.ProcessClassStorage)
+			Expect(originalPodSpec).NotTo(BeNil())
+			modifiedPodSpec = originalPodSpec.DeepCopy()
+			Expect(modifiedPodSpec.SecurityContext).NotTo(BeNil())
+			if originalPodSpec.SecurityContext.FSGroupChangePolicy != nil {
+				Expect(*originalPodSpec.SecurityContext.FSGroupChangePolicy).NotTo(Equal(corev1.FSGroupChangeOnRootMismatch))
+			}
+			modifiedPodSpec.SecurityContext.FSGroupChangePolicy = &[]corev1.PodFSGroupChangePolicy{corev1.FSGroupChangeOnRootMismatch}[0]
+			initialLogPods = fdbCluster.GetLogPods()
+			initialStatelessPods = fdbCluster.GetStatelessPods()
+			// if we do not set the PodUpdateStrategy to delete, then the pods will get replaced for the spec change,
+			// meaning we cannot test the security context change on non-storage pods unless we use PodUpdateStrategyDelete
+			// (and storage pod use would make the test take ~5min longer)
+			spec := fdbCluster.GetCluster().Spec.DeepCopy()
+			initialPodUpdateStrategy = spec.AutomationOptions.PodUpdateStrategy
+			spec.AutomationOptions.PodUpdateStrategy = fdbv1beta2.PodUpdateStrategyDelete
+			fdbCluster.UpdateClusterSpecWithSpec(spec)
+			Expect(fdbCluster.SetPodTemplateSpec(fdbv1beta2.ProcessClassGeneral, modifiedPodSpec, false)).NotTo(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			spec := fdbCluster.GetCluster().Spec.DeepCopy()
+			spec.AutomationOptions.PodUpdateStrategy = initialPodUpdateStrategy
+			fdbCluster.UpdateClusterSpecWithSpec(spec)
+			Expect(fdbCluster.SetPodTemplateSpec(fdbv1beta2.ProcessClassGeneral, originalPodSpec, true)).NotTo(HaveOccurred())
+		})
+
+		It("should replace all the log pods (which had the securityContext change)", func() {
+			lastForcedReconciliationTime := time.Now()
+			forceReconcileDuration := 4 * time.Minute
+			Eventually(func(g Gomega) {
+				// Force a reconcile if needed to make sure we speed up the reconciliation if needed.
+				if time.Since(lastForcedReconciliationTime) >= forceReconcileDuration {
+					fdbCluster.ForceReconcile()
+					lastForcedReconciliationTime = time.Now()
+				}
+
+				// check that all original pods are gone (that use the General template)
+				logPods := fdbCluster.GetLogPods()
+				g.Expect(logPods.Items).NotTo(ContainElements(initialLogPods.Items))
+				for _, pod := range logPods.Items {
+					g.Expect(pod.Spec.SecurityContext.FSGroupChangePolicy).ToNot(BeNil())
+					g.Expect(*pod.Spec.SecurityContext.FSGroupChangePolicy).To(Equal(corev1.FSGroupChangeOnRootMismatch))
+				}
+				statelessPods := fdbCluster.GetStatelessPods()
+				g.Expect(statelessPods.Items).NotTo(ContainElements(initialStatelessPods.Items))
```

Comment:
This seems to always be the case, trying to track down exact changed fields, but I am thinking checking the names / Pgids changed is more meaningful

Reply:
The `fdbCluster.GetPodsNames()` could be used here, e.g.: https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/e2e/test_operator/operator_test.go#L920-L937